src/bloqade/qasm2/parse/lowering.py

```diff
@@ -17,6 +19,118 @@ class QASM2(lowering.LoweringABC[ast.Node]):
     hint_show_lineno: bool = field(default=True, kw_only=True)
     stacktrace: bool = field(default=True, kw_only=True)
 
+    def loads(
+        self,
+        source: str,
+        kernel_name: str,
+        *,
+        returns: str | None = None,
```

Comment:
We can add multiple returns later but for standard qasm2 file you can't do this because we do not support tuples.

Reply:
yeah I think this is ok.